sdk/core/core-amqp/src/requestResponseLink.ts

```diff
@@ -333,3 +337,20 @@ export function onMessageReceived(
   logErrorStackTrace(error);
   return promise.reject(error);
 }
+
+function onSenderError(
+  context: Pick<EventContext, "sender">,
+  connectionId: string,
+  responsesMap: Map<string, DeferredPromiseWithCallback>
+): void {
+  if (context.sender) {
+    for (const [key, promise] of responsesMap.entries()) {
+      logger.verbose(
+        `[${connectionId}] Sender closed due to error when sending request with message_id "${key}"`
+      );
+      promise.cleanupBeforeResolveOrReject();
+      responsesMap.delete(key);
```

Comment:
It makes me nervous to modify a collection inside a loop over its elements. Could `responsesMap.clear()` be called after the loop instead?

Reply:
Updated to clear outside of loop.